core/src/main/java/edu/wpi/grip/core/operations/composite/SaveImageOperation.java

```diff
@@ -0,0 +1,130 @@
+package edu.wpi.grip.core.operations.composite;
+
+import edu.wpi.grip.core.FileManager;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.OperationDescription;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.core.sockets.SocketHints;
+import edu.wpi.grip.core.util.Icon;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableList;
+
+import org.bytedeco.javacpp.BytePointer;
+import org.bytedeco.javacpp.IntPointer;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.bytedeco.javacpp.opencv_core.Mat;
+import static org.bytedeco.javacpp.opencv_imgcodecs.CV_IMWRITE_JPEG_QUALITY;
+import static org.bytedeco.javacpp.opencv_imgcodecs.imencode;
+
+/**
+ * Save JPEG files periodically to the local disk.
+ */
+public class SaveImageOperation implements Operation {
+
+  public static final OperationDescription DESCRIPTION =
+      OperationDescription.builder()
+          .name("Save Images to Disk")
+          .summary("Save image periodically to local disk")
+          .category(OperationDescription.Category.MISCELLANEOUS)
+          .icon(Icon.iconStream("publish-video"))
+          .build();
+
+  private final SocketHint<Mat> inputHint
+      = SocketHints.Inputs.createMatSocketHint("Input", false);
+  private final SocketHint<FileTypes> fileTypeHint
+      = SocketHints.createEnumSocketHint("File type", FileTypes.jpeg);
+  private final SocketHint<Number> qualityHint
+      = SocketHints.Inputs.createNumberSliderSocketHint("Quality", 90, 0, 100);
+  private final SocketHint<Number> periodHint
+      = SocketHints.Inputs.createNumberSpinnerSocketHint("Period", 0.1);
+  private final SocketHint<Boolean> activeHint
+      = SocketHints.Inputs.createCheckboxSocketHint("Active", false);
+
+  private final SocketHint<Mat> outputHint = SocketHints.Outputs.createMatSocketHint("Output");
+
+  private final InputSocket<Mat> inputSocket;
+  private final InputSocket<FileTypes> fileTypesSocket;
+  private final InputSocket<Number> qualitySocket;
+  private final InputSocket<Number> periodSocket;
+  private final InputSocket<Boolean> activeSocket;
+
+  private final OutputSocket<Mat> outputSocket;
+
+  private final FileManager fileManager;
+  private final BytePointer imagePointer = new BytePointer();
+  private final Stopwatch stopwatch = Stopwatch.createStarted();
+  private final DateTimeFormatter formatter
+      = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm-ss-SSS");
+
+  private enum FileTypes {
+    jpeg, png
```

Comment:
Usually enums are all caps. If you want to have the spinner value show up differently then override `toString` and it should work.